src/Microsoft.AspNet.Mvc.Core/ModelBindingHelper.cs

```diff
@@ -530,5 +524,189 @@ public static object ConvertValuesToCollectionType<T>(Type modelType, IList<T> v
+        /// </returns>
+        public static T ConvertTo<T>(object value, CultureInfo culture)
+        {
+            var converted = ConvertTo(value, typeof(T), culture);
+            return converted == null ? default(T) : (T)converted;
+        }
+
+        /// <summary>
+        /// Converts the provided <paramref name="value"/> to a value of <see cref="Type"/> <param name="type"/>
+        /// using the <see cref="CultureInfo.InvariantCulture"/>.
+        /// </summary>
+        /// <param name="value">The value to convert."/></param>
+        /// <param name="type">The <see cref="Type"/> for conversion.</param>
+        /// <returns>
+        /// The converted value or <c>null</c> if the value could not be converted.
+        /// </returns>
+        public static object ConvertTo(object value, [NotNull] Type type)
+        {
+            return ConvertTo(value, type, culture: null);
+        }
+
+        /// <summary>
+        /// Converts the provided <paramref name="value"/> to a value of <see cref="Type"/> <paramref name="type"/>.
+        /// </summary>
+        /// <param name="value">The value to convert."/></param>
+        /// <param name="type">The <see cref="Type"/> for conversion.</param>
+        /// <param name="culture">The <see cref="CultureInfo"/> for conversion.</param>
+        /// <returns>
+        /// The converted value or <c>null</c> if the value could not be converted.
+        /// </returns>
+        public static object ConvertTo(object value, [NotNull] Type type, CultureInfo culture)
+        {
+            if (value == null)
+            {
+                // For value types, treat null values as though they were the default value for the type.
+                return type.GetTypeInfo().IsValueType ? Activator.CreateInstance(type) : null;
+            }
+
+            if (value.GetType().IsAssignableFrom(type))
+            {
+                return value;
+            }
+
+            var cultureToUse = culture ?? CultureInfo.InvariantCulture;
+            return UnwrapPossibleArrayType(value, type, cultureToUse);
+        }
+
+        private static object UnwrapPossibleArrayType(object value, Type destinationType, CultureInfo culture)
+        {
+            // array conversion results in four cases, as below
+            var valueAsArray = value as Array;
+            if (destinationType.IsArray)
+            {
+                var destinationElementType = destinationType.GetElementType();
+                if (valueAsArray != null)
+                {
+                    // case 1: both destination + source type are arrays, so convert each element
+                    var converted = (IList)Array.CreateInstance(destinationElementType, valueAsArray.Length);
+                    for (var i = 0; i < valueAsArray.Length; i++)
+                    {
+                        converted[i] = ConvertSimpleType(valueAsArray.GetValue(i), destinationElementType, culture);
```

Comment:
Fortunately there's now no case in our code where this could happen. We use the `string`->scalar and `string[]`->scalar conversions, but not the others.